### PR TITLE
Feature/my 22307 query consignments

### PR DESCRIPTION
**Pull request by reindert-vetter**
A nice solution for https://github.com/myparcelnl/sdk/issues/204

### CODE REVIEW

File: src/Helper/MyParcelCollection.php [hunk 1]
```diff
@@ -625,6 +608,53 @@ public function clearConsignmentsCollection()
...
+     */
+    public static function query(string $apiKey, $parameters): MyParcelCollection
+    {
+        $collection = new static();
+
+        // The field `size` is required to prevent bugs. Think carefully about what
+        // the maximum size should be in your use case. If you want to pick up all
+        // open consignments for example, you would probably want to adjust size to 300.
+        if (empty($parameters['size'])) {
+            throw new MissingFieldException('Field size is requested.');
```
Comment (EdieLemoine) [inline src/Helper/MyParcelCollection.php:634]: Beetje vage zin, misschien `Field "size" is required.`?

File: src/Helper/MyParcelCollection.php [hunk 2]
```diff
@@ -625,6 +608,53 @@ public function clearConsignmentsCollection()
...
+     * @throws \MyParcelNL\Sdk\src\Exception\MissingFieldException
+     * @throws \Exception
+     */
+    public static function query(string $apiKey, $parameters): MyParcelCollection
+    {
+        $collection = new static();
+
+        // The field `size` is required to prevent bugs. Think carefully about what
+        // the maximum size should be in your use case. If you want to pick up all
+        // open consignments for example, you would probably want to adjust size to 300.
```
Comment (EdieLemoine) [inline src/Helper/MyParcelCollection.php:632]: Multiline comment ;)

Comment (reindert-vetter) [reply src/Helper/MyParcelCollection.php:632]: https://wiki.dmp.zone/display/MD/DSR-1%3A+PHP+Standards#DSR-1:PHPStandards-Comments
Dit is gewoon correct.

File: src/Helper/MyParcelCollection.php (hunk 1 above)
Comment (reindert-vetter) [reply src/Helper/MyParcelCollection.php]: Done

File: src/Helper/MyParcelCollection.php (hunk 2 above)
Comment (EdieLemoine) [reply src/Helper/MyParcelCollection.php:632]: Oh. 😟

File: src/Helper/MyParcelCollection.php [hunk 3]
```diff
@@ -625,6 +608,53 @@ public function clearConsignmentsCollection()
...
+            throw new ApiException('Unknown error in MyParcel API response');
+        }
+
+        foreach ($request->getResult()['data']['shipments'] as $shipment) {
+            $consignmentAdapter = new ConsignmentAdapter($shipment, (ConsignmentFactory::createByCarrierId($shipment['carrier_id'])->setApiKey($apiKey)));
+            $collection->addConsignment($consignmentAdapter->getConsignment());
+        }
+
+        return $collection;
+
```
[the remark this answers is not in the document]
Comment (reindert-vetter) [reply src/Helper/MyParcelCollection.php]: done

**Pull request merged**